release_layer.py

```diff
@@ -0,0 +1,175 @@
+import codecs
+import re
+import subprocess
+import tempfile
+import shutil
+import os
+import json
+
+# The following values are used in the documentation, so any change of them requires updates to the documentation.
+LAYER_NAME = 'AWSCodeGuruProfilerPythonAgentLambdaLayerTestDev'
+SUPPORTED_VERSIONS = ['3.6', '3.7', '3.8']
+EXEC_SCRIPT_FILE_NAME = 'codeguru_profiler_lambda_exec'
+
+# We should release in all the regions that lambda layer is supported, not just the ones CodeGuru Profiler Service supports.
+# See this link for supported regions: https://docs.aws.amazon.com/general/latest/gr/lambda-service.html
+LAMBDA_LAYER_SUPPORTED_REGIONS = ['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2',
+                                  'ap-south-1', 'ap-northeast-2', 'ap-southeast-1', 'ap-southeast-2', 'ap-northeast-1',
+                                  'ap-east-1',
+                                  'ca-central-1',
+                                  'eu-central-1', 'eu-west-1', 'eu-west-2', 'eu-west-3', 'eu-north-1', 'eu-south-1',
+                                  'af-south-1', 'me-south-1', 'sa-east-1',
+                                  'cn-north-1', 'cn-northwest-1',
+                                  'us-gov-west-1', 'us-gov-east-1']
+
+# Now we do not release for some of those regions yet:
+#  - China regions are not available through the lambda console: cn-north-1, cn-northwest-1
+#  - Some regions are opt-in, customers have to manually activate them to use so we will wait for customers to ask
+#    for them: me-south-1, eu-south-1, af-south-1, ap-east-1
+#  - US gov regions are also skipped for now: us-gov-west-1, us-gov-east-1
+SKIPPED_REGIONS = ['cn-north-1', 'cn-northwest-1', 'us-gov-west-1', 'us-gov-east-1',
+                   'me-south-1', 'eu-south-1', 'af-south-1', 'ap-east-1']
+REGIONS_TO_RELEASE_TO = sorted(set(LAMBDA_LAYER_SUPPORTED_REGIONS) - set(SKIPPED_REGIONS))
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+def confirm(prompt_str, answer_true='y', answer_false='n'):
+    """
+    Just a manual prompt to ask for confirmation.
+    This gives time for engineers to check the archive we have generated before publishing.
+    """
+    prompt = '%s (%s|%s): ' % (prompt_str, answer_true, answer_false)
+
+    while True:
+        answer = input(prompt).lower()
+        if answer == answer_true:
+            return True
+        elif answer == answer_false:
+            return False
+        else:
+            print('Please enter ' + answer_true + ' or ' + answer_false)
+
+
+def read(*parts):
+    return codecs.open(os.path.join(here, *parts), 'r').read()
+
+
+def find_version(*file_paths):
```

Comment:
I think this deserves a comment, something like
"""
To keep track of which PyPi version corresponds to which layer version we cross reference the PyPi version in the layer description. This function is used to read the PyPi directly from the source code.
"""

Reply:
Yes, I will add it.